relay_test.go

```diff
@@ -1001,3 +1001,98 @@ func TestRelayRaceCompletionAndTimeout(t *testing.T) {
 		wg.Wait()
 	})
 }
+
+func TestRelayArg2OffsetIntegration(t *testing.T) {
+	ctx, cancel := NewContext(testutils.Timeout(time.Second))
+	defer cancel()
+
+	inspector := newFailedRelayHost()
+	opts := testutils.NewOpts().
+		SetRelayOnly().
+		SetRelayHost(inspector)
+
+	ch := testutils.NewServer(t, opts)
+	defer ch.Close()
+
+	client := testutils.NewClient(t, nil /*opts*/)
+	defer client.Close()
+
+	arg2Data := []byte("moe")
+	arg3Data := []byte("auto-tconfig")
+	exLargeArg2 := make([]byte, MaxFrameSize+100)
+
+	tests := []struct {
+		msg         string
+		arg2Data    []byte
+		arg2Flush   bool
+		arg3Data    []byte
+		wantHasMore bool
```

Comment:
can we also validate the start/end offsets, since those should be constant

Reply:
I was thinking to do it, but hesitated due to the hard coded number. Will add the values.